lib/utils.js

```diff
@@ -57,73 +52,62 @@ function isString(key) {
 //  c) first to settle with a falsy value
 // c) is sub-optimal, since it has a specific narrow use-case
 // for save hooks
-function when(items) {
-  items = items || [];
-  var resolveAll, rejectAll,
-      allResolvedPromise = new Promise(function(resolve, reject) {
-        resolveAll = resolve;
-        rejectAll = reject;
-      });
-
-  var remaining = 0,
-      settledWith = [],
-      itemsCopy = Array.isArray(items) ?
-        items.slice() :
-        [items];
-
-  var resolveWith = function(data, index) {
-    settledWith[index] = data;
+export function when (items) {
+  items = items || []
+  let resolveAll, rejectAll
+  const allResolvedPromise = new Promise((resolve, reject) => {
+    resolveAll = resolve
+    rejectAll = reject
+  })
+
+  let remaining = 0
+  const settledWith = []
+  const itemsCopy = Array.isArray(items) ? items.slice() : [items]
+
+  function resolveWith (data, index) {
+    settledWith[index] = data
 
     if (--remaining <= 0) {
-      resolveAll(settledWith);
+      resolveAll(settledWith)
     }
-  };
+  }
 
-  remaining = itemsCopy.length;
+  remaining = itemsCopy.length
 
   if (remaining <= 0) {
-    resolveAll();
-    return allResolvedPromise;
+    resolveAll()
+    return allResolvedPromise
   }
 
-  itemsCopy.forEach(function(item, index) {
-    var promise;
+  itemsCopy.forEach((item, index) => {
+    var promise
```

Comment:
I noticed a few `var`s - was wondering if it is worthwhile to convert these to `let`s, including the one in the for loop?

Maybe we can do a for(const of) loop?